### PR TITLE
Regex improvement on Powershell rules

### PR DESCRIPTION
**Pull request by djlukic**
### Summary of the Pull Request

Hi,

Proposing a regex fix for following rules:

1. 30a8cb77-8eb3-4cfb-8e79-ad457c5a4592
2. df9a0e0e-fedb-4d6c-8668-d765dfc92aa7

### Changelog

Regex condition update for following rules:

- Suspicious Non PowerShell WSMAN COM Provider
- Renamed Powershell Under Powershell Channel

### Example Log Event

```
- <Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
- <System>
  <Provider Name="PowerShell" /> 
  <EventID Qualifiers="0">400</EventID> 
  <Level>4</Level> 
  <Task>4</Task> 
  <Keywords>0x80000000000000</Keywords> 
  <TimeCreated SystemTime="2024-12-01T14:26:10.0000000Z" /> 
  <EventRecordID>3079110</EventRecordID> 
  <Channel>Windows PowerShell</Channel> 
  <Computer>McServer.mcdermott.local</Computer> 
  <Security /> 
  </System>
- <EventData>
  <Data>Available</Data> 
  <Data>None</Data> 
  <Data>NewEngineState=Available PreviousEngineState=None SequenceNumber=37 HostName=RunspaceHost HostVersion=15.0.0.0 HostId=9d15ca57-8227-4b5f-a365-fed71ccb085b EngineVersion=3.0 RunspaceId=5e473451-e6ff-4eba-b8d8-c96c5fe84509 PipelineId= CommandName= CommandType= ScriptName= CommandPath= CommandLine=</Data> 
  </EventData>
  </Event>
```

### Fixed Issues

Above Example of Log Event is beautified view from Event Viewer.

Screenshot 2024-12-27 122655

However after **HostId=9d15ca57-8227-4b5f-a365-fed71ccb085b** there is no **whitespace** but **\r\n\t**. I checked this in many EVTX parsers to be sure and all of them have **\r\n\t**.

**Windows Powershell.evtx** in HxD shows the same.
Screenshot 2024-12-27 122853

### SigmaHQ Rule Creation Conventions

- If your PR adds new rules, please consider following and applying these [conventions](https://github.com/SigmaHQ/sigma-specification/blob/main/sigmahq/)

### CODE REVIEW

File: rules/windows/powershell/powershell_classic/posh_pc_wsman_com_provider_no_powershell.yml [hunk 1]
```diff
@@ -31,7 +31,7 @@ detection:
     filter_main_host_application_null:
         # Note: Since we're using the raw data field to match. There is no easy way to filter out cases where the "HostApplication" field is null (i.e doesn't exist). We're practically forced to use a regex.
         # If you're already mapping and extracting the field, then obviously use that directly.
-        Data|re: 'HostId=[a-zA-Z0-9-]{36} EngineVersion='
+        Data|re: 'HostId=[a-zA-Z0-9-]{36}\r\n\tEngineVersion='
```
Comment (nasbench) [inline rules/windows/powershell/powershell_classic/posh_pc_wsman_com_provider_no_powershell.yml:34]:
```suggestion
        Data|re: 'HostId=[a-zA-Z0-9-]{36}\s+EngineVersion='
```

File: rules/windows/powershell/powershell_classic/posh_pc_renamed_powershell.yml [hunk 2]
```diff
@@ -30,7 +30,7 @@ detection:
     filter_main_host_application_null:
         # Note: Since we're using the raw data field to match. There is no easy way to filter out cases where the "HostApplication" field is null (i.e doesn't exist). We're practically forced to use a regex.
         # If you're already mapping and extracting the field, then obviously use that directly.
-        Data|re: 'HostId=[a-zA-Z0-9-]{36} EngineVersion='
+        Data|re: 'HostId=[a-zA-Z0-9-]{36}\r\n\tEngineVersion='
```
Comment (nasbench) [inline rules/windows/powershell/powershell_classic/posh_pc_renamed_powershell.yml:33]:
```suggestion
        Data|re: 'HostId=[a-zA-Z0-9-]{36}\s+EngineVersion='
```

**Review by nasbench — APPROVED** [review]
My original logs matched accurately on this using a custom parser. Hence changing this to `\s+` should cover all cases.

**Comment by frack113** [comment]
Regex is good to me, only miss to update the modified date.

**Review by frack113 — APPROVED** [review]

**Pull request merged**